icd/CMakeLists.txt

```diff
@@ -75,8 +75,6 @@ if(MSVC)
     target_compile_options(VkICD_mock_icd PRIVATE /bigobj)
     target_compile_definitions(VkICD_mock_icd PRIVATE _CRT_SECURE_NO_WARNINGS)
     target_link_options(VkICD_mock_icd PRIVATE /DEF:${CMAKE_CURRENT_SOURCE_DIR}/${MOCK_ICD_NAME}.def)
-elseif(MINGW)
-    target_sources(VkICD_mock_icd PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/${MOCK_ICD_NAME}.def)
```

Comment:
Remove implication of MINGW support for mock icd. We don't install this by default (See INSTALL_ICD). So we don't need to support this build config.